frontend/src/SpotifyPlaylistComponents/SpotifyPlaylistPage.tsx

```diff
@@ -1,25 +1,54 @@
 import {useEffect, useState} from "react";
 import {SpotifyUserPlaylists} from "../models";
-import {getAllUserPlaylists} from "../api_methods";
+import {addSpotifyPlaylist, getAllUserPlaylists} from "../api_methods";
 import SpotifyPlaylistComponent from "./SpotifyPlaylistComponent";
 
 
 export default function SpotifyPlaylistPage() {
 
+    const [newPlaylistName, setNewPlaylistName] = useState("")
+    const [newPlaylistDescription, setNewPlaylistDescription] = useState("")
+    const [newPlaylistIsPublicStatus, setNewPlaylistIsPublicStatus] = useState(false)
+    const [publicStatusDisabled, setPublicStatusDisabled] = useState(false)
+    const [newPlaylistIsCollaborativeStatus,setNewPlaylistIsCollaborativeStatus] = useState(false)
     const [allPlaylistsObject, setAllPlaylistsObject] = useState<SpotifyUserPlaylists>()
 
     useEffect(() => {
         getAllUserPlaylists()
             .then(data => setAllPlaylistsObject(data))
     }, [])
 
-    console.log(allPlaylistsObject)
-    const playlists = allPlaylistsObject?.items.map(playlist => <SpotifyPlaylistComponent key={playlist.name} playlist={playlist}/>)
+    useEffect(() => {
+        if(newPlaylistIsCollaborativeStatus) {
+            setNewPlaylistIsPublicStatus(false)
+            setPublicStatusDisabled(true)
+        } else {
+            setNewPlaylistIsPublicStatus(false)
+            setPublicStatusDisabled(false)
+        }
+    }, [newPlaylistIsCollaborativeStatus])
 
+    function addPlaylist() {
+        addSpotifyPlaylist(newPlaylistName, newPlaylistDescription, newPlaylistIsPublicStatus, newPlaylistIsCollaborativeStatus)
+            .catch()
+    }
+
+    const playlists = allPlaylistsObject?.items.map(playlist => <SpotifyPlaylistComponent key={playlist.name} playlist={playlist}/>)
 
     return (
         <div>
-            {playlists}
+            <form onSubmit={addPlaylist}>
+                <input placeholder={"name"} value={newPlaylistName} onChange={ev => setNewPlaylistName(ev.target.value)}/>
+                <input placeholder={"description"} value={newPlaylistDescription} onChange={ev => setNewPlaylistDescription(ev.target.value)}/>
+                <label>Public: </label>
+                <input id={"publicStatus"} type={"checkbox"} checked={newPlaylistIsPublicStatus} disabled={publicStatusDisabled} onChange={ev => setNewPlaylistIsPublicStatus(ev.target.checked)}/>
+                <label>Collaborative: </label>
+                <input type={"checkbox"} checked={newPlaylistIsCollaborativeStatus} onChange={ev => setNewPlaylistIsCollaborativeStatus(ev.target.checked)}/>
+                <button type={"submit"}>Add New Playlist</button>
```

Comment:
nice catch for your frontend: 
If you use conditional rendering for this button, so it's only shown when your inputs are filled, the form will be locked until the button is rendered. So you can already check is all inputs are filled and block submit of an incorrect filled form

Reply:
nice. on my refactor list. tomorrow might be refactoring day